### DISCUSSION TITLE
Google Photos photoTakenTime not imported

### DISCUSSION THREAD

**Discussion by Censored3** [photoprism/photoprism#2124] [Q&A] [answered]
Hello

I'm using PhotoPrism to import my Google Photos library.
I had used GPohos to correct photo taken times extensively in the past.
These corrected dates are properly exported in the JSON sidecars, as expected.

However, for the life of me I cannot seem to get these dates imported back into PhotoPrism, even though I understand that this should be the case.

I tried every possible combination I can think of to see if it's just a "priority" issue (maybe the JSON's photoTakenTime is the lowest priority), I went as far as deleting all EXIF data from some jpegs but even then the heuristics still kick in, I cannot get it to refer the JSON's data.

This is a big issue for me as I have a lot of photos that were corrected this way.

At last I thought it could be due to the format in the JSONs and even changed it manually on some test files to the format of `2002:06:06 05:06:00Z` and `2002:06:06 05:06:00+00`, but to no avail.

The weird thing is, a description does get imported so clearly PhotoPrism is parsing the JSON file, at least partly. I can also see them stack up once imported.

I did all my tests on fresh databases (photoprism reset + delete the whole originals folder) and always tried with both import and index.

I could not find any mention of a comparable issue. Am I hitting a bug here, doing something wrong, or is this maybe by design?

Thanks!

I'm running on a pretty standard Docker (-compose) install based off the sample docker-compose.

### Sample json file
This photo was corrected to "6/6/2002" but always get imported as "29/11/2015"

```{
  "title": "20151129_203856.jpg",
  "description": "A picture with a wrong date",
  "imageViews": "5",
  "creationTime": {
    "timestamp": "1471194915",
    "formatted": "14 aug. 2016 17:15:15 UTC"
  },
  "photoTakenTime": {
    "timestamp": "1023339960",
    "formatted": "6 jun. 2002 05:06:00 UTC"
  },
  "geoData": {
    "latitude": 50.793055599999995,
    "longitude": 4.915555599999999,
    "altitude": -0.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "geoDataExif": {
    "latitude": 50.793055599999995,
    "longitude": 4.915555599999999,
    "altitude": -0.0,
    "latitudeSpan": 0.0,
    "longitudeSpan": 0.0
  },
  "url": "https://lh3.googleusercontent.com/<REDACTED>",
  "googlePhotosOrigin": {
    "webUpload": {
      "computerUpload": {
      }
    }
  },
  "photoLastModifiedTime": {
    "timestamp": "1644611042",
    "formatted": "11 feb. 2022 20:24:02 UTC"
  }
}
```

one more possibly useful thing I found in the log at import:
```
photoprism_1  | time="2022-03-05T11:42:57Z" level=debug msg="metadata: exif not supported in 20151129_203856.jpg.json"
```

**Comment by graciousgrey** [comment]
@Censored3 Could you send us examples (the photo + related json) directly from your Google Photos Takeout for debugging to samples@photoprism.app?

**Comment by Censored3** [comment]
Certainly! I've sent some over right now.

**Reply by graciousgrey** [reply]
Thank you! We will have a look within the next weeks.

**Comment by graciousgrey** [comment, marked as answer]
The problem here is that the wrong date information is still present in your files exif data. 

If the wrong exif dates are removed before indexing e.g. using

`exiftool -CreateDate= -DateTimeOriginal= -GPSDateTime= -GPSDateStamp= -ModifyDate= filename.jpg`,

the dates from the json are used.

I created a ticket to prioritize the date  from the json over the ones from exif (https://github.com/photoprism/photoprism/issues/2133).

**Reply by Censored3** [reply]
So I must've missed something in my testing.
Thanks a lot foor looking into this - looking forward to start using PP!
